### PR TITLE
Travis: optimization

### PR DESCRIPTION
**Pull request by maxGimeno**
## Summary of Changes
Moves the dependencies test into the install: step to gain more time if a package is not concerned.

### CODE REVIEW

**Comment by lrineau** [comment]
https://travis-ci.org/CGAL/cgal/requests
>  #2992 c21fad9
> 19 minutes ago
> Try to move the test_package step into -install:
> - Could not parse .travis.yml

**Comment by lrineau** [comment]
same

**Comment by maxGimeno** [comment]
Ok I found the right syntax. I have to put back the test in build_package as $PACKAGE groups the packages by 3.

File: .travis/test_package.sh [hunk 1]
```diff
@@ -5,12 +5,12 @@
 
 DO_IGNORE=FALSE
 cd $1
-if [ ! -f "$2/package_info/$ARG/dependencies" ];then 
+if [ ! -f "$2/package_info/$2/dependencies" ];then 
   echo "No dependencies found for $2"
-  return
+  exit 0
```
[the remark this answers is not in the document]
Comment (maxGimeno) [reply .travis/test_package.sh]: Indeed

**Comment by maxGimeno** [comment]
Considering the absence of dependencies as an error showed that MacOSX, Scripts and  wininst have no dependencies file. I added an empty one.

**Comment by lrineau** [comment]
The new timings are much better (here with a PR modifying only `Documentation/`:
https://travis-ci.org/CGAL/cgal/builds/364996371

The time is fully spent in the CHECK part, now. I suggest:
- to remove the CHECK with gcc (and keep only clang)
- ... we should try a way to check headers only for modified packages (and packages depending on them).

**Comment by maxGimeno** [comment]
We currently check that the dependencies are up to date in CHECK. Restricting the CHECK to the modified packages would require to drop this test, but we now heavily rely on the dependencies in Travis. I don't think we should stop testing them.

File: .travis/template.txt [hunk 2]
```diff
@@ -7,6 +7,8 @@ matrix:
  include:
   PACKAGES_MATRIX
 install: 
+  - echo "$PWD"
+  - if [ -n "$TRAVIS_PULL_REQUEST" ] && [ "$PACKAGE" != CHECK ]; then  DO_IGNORE=FALSE;  for ARG in $(echo "$PACKAGE");do . $PWD/.travis/test_package.sh "$PWD" "$ARG";  echo "DO_IGNORE is $DO_IGNORE";  if [ "$DO_IGNORE" = "TRUE" ]; then  travis_terminate 0; fi; done;fi
```
Comment (lrineau) [inline .travis/template.txt]: That logic is wrong, in `.travis.yml`. See [here for a PR modifying Triangulation_2: Triangulation_3 can be ignored, but the full test of "Triangulation_2 Triangulation_3 Union_find" is ignored!][1]

[1]: https://travis-ci.org/CGAL/cgal/jobs/380773569

**Pull request merged**